pyanaconda/modules/subscription/subscription.py

```diff
@@ -844,6 +845,42 @@ def _unregister_task_callback(self, task_instance):
             self.set_subscription_attached(False)
             self.clear_subscription_data()
 
+    def _get_subscription_state_data(self):
+        """Get data from RHSM describing what subscriptions have been attached to the system.
+
+        Calling AutoAttach() also generally returns such data, but due to bug 1790924,
+        we can't depend on it always being the case.
+
+        Therefore, we query subscription state separately using the GetPools() method.
+        """
+
+        locale = os.environ.get("LANG", "")
+        # fetch subscription status data
+
+        subscription_json = entitlement_proxy.GetPools(
+                {"pool_subsets": get_variant(Str, "consumed")},
+                {},
+                locale
+        )
+        subscription_data_length = 0
+        # Log how much subscription data we got for debugging purposes.
+        # By only logging length, we should be able to debug cases of no
+        # or incomplete data being logged, without logging potentially
+        # sensitive subscription status detail in the installation logs stored
+        # on the target system.
+        if subscription_json:
+            subscription_data_length = len(subscription_json)
+        log.debug("RHSM: fetched subscription status data: %d characters",
```

Comment:
I think you should move this under the `if` otherwise you will log `0`.

Reply:
This is actually intended, so that we detect if nothing was returned by the API. But I guess a can add a separate warning level log message for that. :)